src/main/java/com/pidoco/juri/JURI.java

```diff
@@ -974,6 +995,67 @@ public JURI replaceQueryParametersMulti(Map<String, Collection<String>> params)
         return this;
     }
 
+    /**
+     * Navigate to a new URI.<br>
+     * Changes the current JURI inplace, does not create a new object.<br>
+     * If the path is altered, it is normalized.<br>
+     * <br>
+     * Examples:
+     * <ul>
+     *   <li>"http://example.com/a/b.html".navigate("http://www.google.com/search?q=2") => "http://www.google.com/search?q=2"</li>
+     *   <li>"http://example.com/a/b.html".navigate("c.html") => "http://example.com/a/c.html"</li>
+     *   <li>"http://example.com/a/b.html".navigate("../c.html") => "http://example.com/c.html"</li>
+     *   <li>"http://example.com/a/b.html".navigate("../../../../c.html") => "http://example.com/c.html"</li>
+     *   <li>"http://example.com/a/b.html".navigate("#anchor") => "http://example.com/a/b.html#anchor"</li>
+     * </ul>
+     *
+     * @param path May be a relative path or a absolute URI to navigate to or an url fragment. 
+     * @return this.
+     */
+    public JURI navigate(String path) {
+        JURI newURI = JURI.parse(path);
+        if (newURI.getHost() != null) {
+            return this.resetBlank()
+                    .parseNew(path);
+        }
+
+        String newPath = newURI.getPath();
+        Map<String, Collection<String>> newQuery = newURI.getQueryParameters();
+        String newFragment = newURI.getFragment();
+
+        if (newURI.isHavingPath()) {
+            // set new path (and maybe query and fragment):
+            if (newURI.isPathRelative()) {
+                // relative path:
+                // calculate new normalized absolute path
+                newPath = this.clone()
+                        .addRawPath("../" + newURI.getPath())
+                        .getCurrentUri()
+                        .normalize()
+                        .getPath();
+                while (newPath.startsWith("/../")) {
+                    newPath = newPath.substring(3);
+                }
+            }
+            this.setPath(newPath)
+                    .clearQueryParameters()
```

Comment:
If the new query doesn't have query params and the old had they will get deleted. Probably the right thing to do if path changes - but also against intuition at least for me. Maybe add an example to documentation?